### ISSUE TITLE
Wake eyesome on your own via command line

### ISSUE THREAD

**Issue by AdiWeit** [WinEunuuchs2Unix/eyesome#11]
system info: 
oerating system: Linux Mint 20.1 Cinnamon
Cinnemon-Version: 4.8.6
Linux-Kernal: 5.4.0-84-generic

I wrote a script to turn off the screen of my laptop (external still on): `xrandr --output eDP-1 --off`. 
Unfortunately that command resets the brightness to 100% as well, so it would be great to add a command to wake eyesome just like it does automaticly when I close the lid.
I already run the script `chmod +x wake-eyesome.sh` and it seems that it was successful, but it doesn't  seem to do anything when I run `wake-eyesome.sh` or `sh [location]/wake-eyesome.sh`

Alternatively, a better way to turn off the screen would help too. If possible with also keeping the work spaces of that screen. 
When I turn the brightness down to 0% (with keys, just effects laptop) it of cause is black, but when I then black out all screens with a command and later use the laptop again, it turns the screen brightness (laptop) back to 100%

**Comment by WinEunuuchs2Unix** [comment]
@AdiWeit `wake-eyesome.sh` takes parameters when you call it. There is a script I wrote in AskUbuntu.com called `movie.sh` that blacks out the laptop display and disables eyesome for the laptop and 4K TV, leaving only the big screen TV turned on. You can see it [here](https://askubuntu.com/questions/1155093/turn-off-all-monitors-while-watching-vlc-media-on-tv/1155095#1155095).

The relevant code in `movie.sh` for disabling (and then re-enabling) eyesome for some monitors is:

```

DisableAutoBright () {
    # Disable specific monitors from nightlight, redshift, flux, eyesome, ect.
    [[ $CFG_FILE == "" ]] && return # No third party apps
    [[ $MON1 != "" ]] && sed -i 's/1|Enabled/1|Disabled/g' "$CFG_FILE"
    [[ $MON2 != "" ]] && sed -i 's/2|Enabled/2|Disabled/g' "$CFG_FILE"
    [[ $MON3 != "" ]] && sed -i 's/3|Enabled/3|Disabled/g' "$CFG_FILE"
} # DisableAutoBright

EnableAutoBright () {
    # Renable specific monitors for nightlight, redshift, flux, eyesome, ect.
    [[ $CFG_FILE == "" ]] && return # No third party apps
    [[ $MON1 != "" ]] && sed -i 's/1|Disabled/1|Enabled/g' "$CFG_FILE"
    [[ $MON2 != "" ]] && sed -i 's/2|Disabled/2|Enabled/g' "$CFG_FILE"
    [[ $MON3 != "" ]] && sed -i 's/3|Disabled/3|Enabled/g' "$CFG_FILE"

    # Trigger 3rd party app to gradually set brightness based on sunrise/sunset
    /usr/local/bin/wake-eyesome.sh post eyesome-cfg.sh nosleep

} # EnableAutoBright```
```

A very important note is the last command line shows you how to call `wake-eyesome` with the appropriate parameters of `post eyesome-cfg.sh nosleep` which it sounds like you weren't doing earlier.

Another note is if you power off your system when eyesome is disabled for monitor(s) then you must use `eyesome-cfg.sh` to manually enable the monitor(s).

I hope this helps! If not don't hesitate to ask more questions. Thanks.

**Comment by AdiWeit** [comment]
The command works, so thanks a lot for that!
So I run the code and eyesome activated, but now it seems like it turned the night mode (wakes up every 60 seconds and ) on, although it's 12:00 already.
before running the code I entered the website https://www.timeanddate.com/sun/germany/soest for better accuracy and in the setting it then said: 

> Sunrise setting: 07:40span class="com
> 18:49span class="com

, so it could have to do with some sort of wrong scraping?
Otherwise overwriting it manually should work too, right?
my general settings:
![image](https://user-images.githubusercontent.com/43521682/136543871-9a9d777b-b7cb-435c-ad7c-49eda0ab4588.png)


I also don't quiet understand the override functionality. It even says 

> (pause eyesome demon)

behind the name of the functionality when you clicked on the button (to the right of the night symbol), but I don't understand how to do that. 
As far as I understand you can get the currently active settings, change them and apply them, but when I hit "apply" it asks me if I want to change the settings in the configurations file, so I think it will change them permanently. . 

PS: I don't know if you understood me right, but I ment an external screen connected to the laptop I use for working on the laptop (Linux), but the code snippet helped anyway, so thanks for that!

**Comment by WinEunuuchs2Unix** [comment]
@AdiWeit  The bottom check box on your screen can be checked on for: "Watch external monitor plugging/power switching" this may help the switching screens between laptop.

As far as the sunrise and sunset times on my system,  `www.timeanddate.com/sun/germany/soest` shows:

![germany sunrise sunset](https://user-images.githubusercontent.com/33849503/136669327-bc23e807-b428-4da0-a9dc-2f31f5784419.png)

On your system I think it is configured for 24 hour clock and not 12 hour clock like mine? Therefore it is showing "18:46pm" on yours and "6:46pm" on mine. Please confirm.

**Comment by AdiWeit** [comment]
So my system is configures for 24 hour clock, but in the program (eyesome) it sais that it is am/pm format.
On the website it has the 24 hour clock for me. The 

> Sunrise setting: 07:40span class="com
> Sunset setting: 18:49span class="com

was written in eyesome itself in the sun times settings.

The checkbox does seem to work but has a bit of delay. No issue if you know it. :+1:

**Comment by WinEunuuchs2Unix** [comment]
Eyesome is designed around a 12 hour clock with `am` and `pm` designation.  You could manually update the files seen here:

```
$ cat /usr/local/bin/sunrise

7:12 am

$ cat /usr/local/bin/sunset

7:41 pm
```

I've made a mental note to upgrade `eyesome.sh` (or more likely `eyesome-sun.sh` to support 24-hour clock in the future. Thank you for reporting the problem.

I know about the delay resetting brightness when a monitor is re-plugged or power cycled. This is because `dbus` is polled for the interface `"org.freedesktop.ColorManager"` and it has a delay that cannot be changed. You can see the bash code in `/usr/local/bin/eyesome-dbus.sh`.

**Comment by AdiWeit** [comment]
I think the issue is that the sunset hours (at least in the settings can only be set to 1-12) are applied to am as well. So when I set sunset hour to 6:38 it turned the night mode on because it was way later than 6:38 already. 
When I set it to 12 on the other hand (right now it's 8:18) it turned the day mode on. 
In the morning when I rebooted the PC it still had my (in the settings) overwritten settings, but when seyesome demon waked because of the sunrise ( the timing could have to do with my overwriting timing as well if it lasts 24h) it changed the settings again (manuel sunset hour automaticly pushed from 6 to 12. I think `mattharbord` meant the same issure, because he described the sunset time jump up to 12 as well.): 

![image](https://user-images.githubusercontent.com/43521682/137078380-aace7bfd-5710-4f45-a72d-5df8e2e86d9b.png)

I still have the eyesome folder in the downloads folder, but that shouldn't be a problem, right?
I also changed the files already, but forgot to break the URL, so that's the reason it reset it I think. 
So now I broke the URL by putting 

> broken: https://www.timeanddate.com/astronomy/germany/soest/

into the text box and changed the sunset time to 18:38 in the file. 

I'll report how it goes!
